<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Browsing the elements of the object â€¢ bcbioSmallRna</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Browsing the elements of the object">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSmallRna</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/browsing-object.html">Browsing the elements of the object</a>
    </li>
    <li>
      <a href="../articles/qc-example.html">QC for small RNA-seq data analyzed with bcbio</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lpantano/bcbiosRnaseq">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Browsing the elements of the object</h1>
                        <h4 class="author">Lorena Pantano</h4>
            
            <h4 class="date">4 October 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lpantano/bcbiosRnaseq/blob/master/vignettes/browsing-object.Rmd"><code>vignettes/browsing-object.Rmd</code></a></small>
      <div class="hidden name"><code>browsing-object.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>bcbioSmallRna package version: 0.0.1</p>
    </div>
    
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BiocStyle)
knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">tidy=</span><span class="ot">FALSE</span>,
                      <span class="dt">dev=</span><span class="st">"png"</span>,
                      <span class="dt">message=</span><span class="ot">FALSE</span>, <span class="dt">error=</span><span class="ot">FALSE</span>,
                      <span class="dt">warning=</span><span class="ot">TRUE</span>)
<span class="kw">library</span>(knitr)
<span class="kw">library</span>(ggplot2)

<span class="co"># Set seed for reproducibility</span>
<span class="kw">set.seed</span>(1454944673L)

<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme_get">theme_set</a></span>(
    <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/ggtheme">theme_light</a></span>(<span class="dt">base_size =</span> 11L))
<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme_get">theme_update</a></span>(
    <span class="dt">legend.justification =</span> <span class="st">"center"</span>,
    <span class="dt">legend.position =</span> <span class="st">"bottom"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(isomiRs)
<span class="kw">library</span>(DEGreport)
<span class="kw">library</span>(bcbioSmallRna)

<span class="kw">data</span>(sbcb)
<span class="co"># bcbioSmallRnaDataSet</span>
bcb &lt;-<span class="st"> </span>sbcb</code></pre></div>
<div id="get-count-matrix" class="section level1">
<h1 class="hasAnchor">
<a href="#get-count-matrix" class="anchor"></a>Get count matrix</h1>
<p>You can get all the count matrix with the method <code>mirna</code>, <code>isomir</code>, <code>cluster</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># for miRNAs</span>
<span class="kw">head</span>(<span class="kw"><a href="../reference/mirna.html">mirna</a></span>(bcb))</code></pre></div>
<pre><code>##               ERR187490 ERR187494 ERR187664 ERR187665
## hsa-let-7a-3p        26       102        25       197
## hsa-let-7a-5p     15396     88290     30838    111189
## hsa-let-7b-3p         8         0         0        39
## hsa-let-7b-5p       400       229       106      1067
## hsa-let-7c-5p        58        58        93       115
## hsa-let-7d-3p       124       560       265       848</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># for clusters</span>
<span class="kw">head</span>(<span class="kw"><a href="../reference/mirna.html">cluster</a></span>(bcb))</code></pre></div>
<pre><code>##           ERR187490 ERR187494 ERR187664 ERR187665
## cluster:1        32       190       126        55
## cluster:2      1033      6675      2283      8369
## cluster:3       313      1077       564      2132
## cluster:4      2996     16959     15000     48050
## cluster:5       465      2470       729      2259
## cluster:6         9        43        14        47</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># for isomir</span>
<span class="kw">head</span>(<span class="kw"><a href="../reference/mirna.html">isomir</a></span>(bcb))</code></pre></div>
<pre><code>##                         ERR187490 ERR187494 ERR187664 ERR187665
## hsa-let-7a-3p                   5        26         5        36
## hsa-let-7a-3p;iso_3p:c          3         8         3        27
## hsa-let-7a-3p;iso_3p:C          0         2         0         3
## hsa-let-7a-3p;iso_3p:tc         3         0         0         0
## hsa-let-7a-3p;iso_add:A         0         4         0         6
## hsa-let-7a-3p;iso_add:T        15        54        17       116</code></pre>
<p>By default this is the raw count data, however you can access a pre-computed normalized data using the second positional parameter <code>log</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/mirna.html">mirna</a></span>(bcb, <span class="st">"log"</span>))</code></pre></div>
<pre><code>##               ERR187490 ERR187494 ERR187664 ERR187665
## hsa-let-7a-3p  6.215516  6.219226  5.698424  6.803659
## hsa-let-7a-5p 15.086041 15.638169 15.467636 15.723060
## hsa-let-7b-3p  5.021667  3.046144  3.046144  5.063780
## hsa-let-7b-5p  9.844976  7.212310  7.424764  9.064181
## hsa-let-7c-5p  7.199906  5.603116  7.254332  6.159710
## hsa-let-7d-3p  8.210866  8.408052  8.663905  8.744015</code></pre>
</div>
<div id="metrics" class="section level1">
<h1 class="hasAnchor">
<a href="#metrics" class="anchor"></a>Metrics</h1>
<p>There are some important metris stored in the object that can be gotten with the following methods:</p>
<div id="adapter-removal" class="section level2">
<h2 class="hasAnchor">
<a href="#adapter-removal" class="anchor"></a>Adapter removal</h2>
<p>These section shows how to get general stats for the adapter removal step.</p>
<p>To get the numbers of adapters removed at each position:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/adapter.html">adapter</a></span>(bcb)[[<span class="st">"reads_by_pos"</span>]])</code></pre></div>
<pre><code>##   size  reads    sample colorby
## 1   17 155324 ERR187490 BRITISH
## 2   18 293195 ERR187490 BRITISH
## 3   19 155948 ERR187490 BRITISH
## 4   20 187603 ERR187490 BRITISH
## 5   21 211411 ERR187490 BRITISH
## 6   22 338768 ERR187490 BRITISH</code></pre>
<p>As well, the total reads with adapter can be seen with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/adapter.html">adapter</a></span>(bcb)[[<span class="st">"reads_by_sample"</span>]]</code></pre></div>
<pre><code>## # A tibble: 4 x 3
## # Groups:   sample [?]
##   sample    colorby   total
##   &lt;chr&gt;     &lt;fct&gt;     &lt;int&gt;
## 1 ERR187490 BRITISH 2457059
## 2 ERR187494 FINLAND 6048597
## 3 ERR187664 USA     3759076
## 4 ERR187665 NIGERIA 5772822</code></pre>
</div>
<div id="general-metrics" class="section level2">
<h2 class="hasAnchor">
<a href="#general-metrics" class="anchor"></a>General metrics</h2>
<p>All the metrics performed by bcbio can be seen with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/metrics.html">metrics</a></span>(bcb)</code></pre></div>
<pre><code>##   country  group    sample library_size quality_format read_pass_filter
## 1 BRITISH group1 ERR187490           30       standard          8594767
## 2 FINLAND group1 ERR187494           30       standard         11802968
## 3     USA group2 ERR187664           30       standard          9697283
## 4 NIGERIA group2 ERR187665           40       standard          8176320
##   read_with_adapter reads_before_trimming sequence_length
## 1           3098670               8594767           17-28
## 2           8488581              11802968           17-28
## 3           4334146               9697283           17-28
## 4           8075701               8176320           17-42
##   sequences_flagged_as_poor_quality x_gc
## 1                                 0   51
## 2                                 0   51
## 3                                 0   49
## 4                                 0   49</code></pre>
</div>
</div>
<div id="session" class="section level1">
<h1 class="hasAnchor">
<a href="#session" class="anchor"></a>Session</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.3
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] bcbioSmallRna_0.0.1         DEGreport_1.17.5           
##  [3] isomiRs_1.9.1               SummarizedExperiment_1.10.1
##  [5] DelayedArray_0.6.5          BiocParallel_1.14.2        
##  [7] matrixStats_0.54.0          Biobase_2.40.0             
##  [9] GenomicRanges_1.32.6        GenomeInfoDb_1.16.0        
## [11] IRanges_2.14.11             S4Vectors_0.18.3           
## [13] BiocGenerics_0.26.0         DiscriMiner_0.1-29         
## [15] ggplot2_3.0.0               knitr_1.20                 
## [17] BiocStyle_2.8.2            
## 
## loaded via a namespace (and not attached):
##   [1] assertive.base_0.0-7        colorspace_1.3-2           
##   [3] rjson_0.2.20                rprojroot_1.3-2            
##   [5] circlize_0.4.4              htmlTable_1.12             
##   [7] XVector_0.20.0              ggdendro_0.1-20            
##   [9] GlobalOptions_0.1.0         base64enc_0.1-3            
##  [11] fs_1.2.6                    rstudioapi_0.7             
##  [13] roxygen2_6.1.0              assertive.sets_0.0-3       
##  [15] MultiAssayExperiment_1.6.0  ggrepel_0.8.0              
##  [17] bit64_0.9-7                 fansi_0.3.0                
##  [19] AnnotationDbi_1.42.1        xml2_1.2.0                 
##  [21] splines_3.5.1               logging_0.7-103            
##  [23] mnormt_1.5-5                geneplotter_1.58.0         
##  [25] Formula_1.2-3               Nozzle.R1_1.1-1            
##  [27] broom_0.5.0                 annotate_1.58.0            
##  [29] cluster_2.0.7-1             readr_1.1.1                
##  [31] compiler_3.5.1              backports_1.1.2            
##  [33] assertthat_0.2.0            Matrix_1.2-14              
##  [35] lazyeval_0.2.1              cli_1.0.0                  
##  [37] limma_3.36.3                lasso2_1.2-19              
##  [39] acepack_1.4.1               htmltools_0.3.6            
##  [41] tools_3.5.1                 bindrcpp_0.2.2             
##  [43] gtable_0.2.0                glue_1.3.0                 
##  [45] GenomeInfoDbData_1.1.0      dplyr_0.7.6                
##  [47] Rcpp_0.12.18                pkgdown_1.1.0              
##  [49] gdata_2.18.0                nlme_3.1-137               
##  [51] psych_1.8.4                 stringr_1.3.1              
##  [53] gtools_3.8.1                XML_3.98-1.16              
##  [55] edgeR_3.22.3                zlibbioc_1.26.0            
##  [57] MASS_7.3-50                 scales_1.0.0               
##  [59] hms_0.4.2                   RColorBrewer_1.1-2         
##  [61] ComplexHeatmap_1.18.1       yaml_2.2.0                 
##  [63] memoise_1.1.0               gridExtra_2.3              
##  [65] rpart_4.1-13                reshape_0.8.7              
##  [67] latticeExtra_0.6-28         stringi_1.2.4              
##  [69] RSQLite_2.1.1               genefilter_1.62.0          
##  [71] desc_1.2.0                  checkmate_1.8.5            
##  [73] caTools_1.17.1.1            shape_1.4.4                
##  [75] rlang_0.2.2                 pkgconfig_2.0.2            
##  [77] commonmark_1.5              bitops_1.0-6               
##  [79] evaluate_0.11               lattice_0.20-35            
##  [81] purrr_0.2.5                 bindr_0.1.1                
##  [83] htmlwidgets_1.2             cowplot_0.9.3              
##  [85] bit_1.1-14                  tidyselect_0.2.4           
##  [87] GGally_1.4.0                plyr_1.8.4                 
##  [89] magrittr_1.5                DESeq2_1.20.0              
##  [91] R6_2.2.2                    gplots_3.0.1               
##  [93] Hmisc_4.1-1                 DBI_1.0.0                  
##  [95] pillar_1.3.0                foreign_0.8-71             
##  [97] withr_2.1.2                 survival_2.42-6            
##  [99] RCurl_1.95-4.11             nnet_7.3-12                
## [101] tibble_1.4.2                janitor_1.1.1              
## [103] crayon_1.3.4                utf8_1.1.4                 
## [105] KernSmooth_2.23-15          rmarkdown_1.10             
## [107] GetoptLong_0.1.7            locfit_1.5-9.1             
## [109] grid_3.5.1                  data.table_1.11.4          
## [111] blob_1.1.1                  ConsensusClusterPlus_1.44.0
## [113] digest_0.6.16               xtable_1.8-3               
## [115] tidyr_0.8.1                 munsell_0.5.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#get-count-matrix">Get count matrix</a></li>
      <li>
<a href="#metrics">Metrics</a><ul class="nav nav-pills nav-stacked">
<li><a href="#adapter-removal">Adapter removal</a></li>
      <li><a href="#general-metrics">General metrics</a></li>
      </ul>
</li>
      <li><a href="#session">Session</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Lorena Pantano, Michael J. Steinbaugh, Harvard Chan Bioinformatics Core.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
