<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>QC for small RNA-seq data analyzed with bcbio • bcbioSmallRna</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="QC for small RNA-seq data analyzed with bcbio">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bcbioSmallRna</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/qc-example.html">QC for small RNA-seq data analyzed with bcbio</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/lpantano/bcbiosRnaseq">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>QC for small RNA-seq data analyzed with bcbio</h1>
                        <h4 class="author">Lorena Pantano</h4>
            
            <h4 class="date">6 June 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/lpantano/bcbiosRnaseq/blob/master/vignettes/qc-example.Rmd"><code>vignettes/qc-example.Rmd</code></a></small>
      <div class="hidden name"><code>qc-example.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>bcbioSmallRna package version: 0.0.1</p>
    </div>
    
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(knitr)
<span class="kw">library</span>(ggplot2)

<span class="co"># Set seed for reproducibility</span>
<span class="kw">set.seed</span>(1454944673L)

<span class="kw">library</span>(BiocStyle)
knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">tidy=</span><span class="ot">FALSE</span>,
                      <span class="dt">dev=</span><span class="st">"png"</span>,
                      <span class="dt">message=</span><span class="ot">FALSE</span>, <span class="dt">error=</span><span class="ot">FALSE</span>,
                      <span class="dt">warning=</span><span class="ot">TRUE</span>)

<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme_get">theme_set</a></span>(
    <span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/ggtheme">theme_light</a></span>(<span class="dt">base_size =</span> 11L))
<span class="kw"><a href="http://www.rdocumentation.org/packages/ggplot2/topics/theme_get">theme_update</a></span>(
    <span class="dt">legend.justification =</span> <span class="st">"center"</span>,
    <span class="dt">legend.position =</span> <span class="st">"bottom"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(isomiRs)
<span class="kw">library</span>(DEGreport)
<span class="kw">library</span>(bcbioSmallRna)
<span class="kw">library</span>(ComplexHeatmap)
<span class="kw">library</span>(circlize)

<span class="kw">data</span>(sbcb)
<span class="co"># bcbioSmallRnaDataSet</span>
bcb &lt;-<span class="st"> </span>sbcb</code></pre></div>
<div id="exploratory-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#exploratory-analysis" class="anchor"></a>Exploratory analysis</h1>
<p>In this section we will see descriptive figures about quality of the data, reads with adapter, reads mapped to miRNAs, reads mapped to other small RNAs.</p>
<div id="size-distribution" class="section level2">
<h2 class="hasAnchor">
<a href="#size-distribution" class="anchor"></a>Size distribution</h2>
<p>After adapter removal, we can plot the size distribution of the small RNAs. We expect the majority of reads to have adapters and we expect a peak at 22 (maybe 33 as well) that indicates miRNA/tRNA enrichment. There are some cases this rule won’t apply.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plots-smallrna.html">bcbSmallSize</a></span>(bcb)</code></pre></div>
<p><img src="qc-example_files/figure-html/adapter-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plots-smallrna.html">bcbSmallSizeDist</a></span>(bcb)</code></pre></div>
<p><img src="qc-example_files/figure-html/adapter-size-dist-1.png" width="700"></p>
</div>
<div id="mirna" class="section level2">
<h2 class="hasAnchor">
<a href="#mirna" class="anchor"></a>miRNA</h2>
<p>A microRNA (abbreviated miRNA) is a small non-coding RNA molecule (containing about 22 nucleotides) found in plants, animals and some viruses, that functions in RNA silencing and post-transcriptional regulation of gene expression.[<a href="https://en.wikipedia.org/wiki/MicroRNA" class="uri">https://en.wikipedia.org/wiki/MicroRNA</a>]</p>
<div id="total-mirna-expression-annotated-with-mirbase" class="section level3">
<h3 class="hasAnchor">
<a href="#total-mirna-expression-annotated-with-mirbase" class="anchor"></a>Total miRNA expression annotated with mirbase</h3>
<p><a href="www.mirbase.org">miRBase</a> is one of the database that contains the reference miRNA sequences. We used this database to annotate the sequences with <a href="http://seqcluster.readthedocs.io/mirna_annotation.html">miraligner</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plots-smallrna.html">bcbSmallMicro</a></span>(bcb)</code></pre></div>
<div class="figure">
<img src="qc-example_files/figure-html/plot_mirna_counts-1.png" alt="Bottom figure tells how many miRNAs until saturation has been detected. Flatten at low value means few miRNAs capture the majority of the coverage." width="700"><p class="caption">
Bottom figure tells how many miRNAs until saturation has been detected. Flatten at low value means few miRNAs capture the majority of the coverage.
</p>
</div>
</div>
<div id="clustering-of-samples" class="section level3">
<h3 class="hasAnchor">
<a href="#clustering-of-samples" class="anchor"></a>Clustering of samples</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data =<span class="st"> </span><span class="kw"><a href="../reference/plotsSmallPCA.html">bcbSmallPCA</a></span>(bcb)
color_by =<span class="st"> </span><span class="kw">metadata</span>(bcb)[[<span class="st">"interesting_groups"</span>]]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">palette &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/circlize/topics/colorRamp2">colorRamp2</a></span>(<span class="kw">seq</span>(<span class="kw">min</span>(data[[<span class="st">"counts"</span>]]),
                          <span class="kw">max</span>(data[[<span class="st">"counts"</span>]]), <span class="dt">length =</span> <span class="dv">3</span>),
                      <span class="kw">c</span>(<span class="st">"blue"</span>, <span class="st">"#EEEEEE"</span>, <span class="st">"orange"</span>), <span class="dt">space =</span> <span class="st">"RGB"</span>)
th &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ComplexHeatmap/topics/HeatmapAnnotation">HeatmapAnnotation</a></span>(<span class="dt">df =</span> data[[<span class="st">"annotation"</span>]],
                        <span class="dt">col =</span> <span class="kw"><a href="http://www.rdocumentation.org/packages/DEGreport/topics/degColors">degColors</a></span>(data[[<span class="st">"annotation"</span>]]))
<span class="kw"><a href="http://www.rdocumentation.org/packages/ComplexHeatmap/topics/Heatmap">Heatmap</a></span>(data[[<span class="st">"counts"</span>]],
        <span class="dt">col =</span> palette,
        <span class="dt">top_annotation =</span> th,
        <span class="dt">clustering_method_rows =</span> <span class="st">"ward.D"</span>,
        <span class="dt">clustering_distance_columns =</span> <span class="st">"kendall"</span>,
        <span class="dt">clustering_method_columns =</span> <span class="st">"ward.D"</span>,
        <span class="dt">show_row_names =</span> <span class="ot">FALSE</span>,
        <span class="dt">show_column_names =</span> <span class="kw">ncol</span>(data[[<span class="st">"counts"</span>]]) <span class="op">&lt;</span><span class="st"> </span><span class="dv">50</span>)</code></pre></div>
<p><img src="qc-example_files/figure-html/mirna-heatmap-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/DEGreport/topics/degPCA">degPCA</a></span>(data[[<span class="st">"counts"</span>]], data[[<span class="st">"annotation"</span>]],
            <span class="dt">condition =</span> color_by)</code></pre></div>
<p><img src="qc-example_files/figure-html/mirna-pca-1.png" width="700"></p>
</div>
</div>
<div id="isomirs" class="section level2">
<h2 class="hasAnchor">
<a href="#isomirs" class="anchor"></a>isomiRs</h2>
<p>isomiR is a term coined by Morin et al. to refer to those sequences that have variations with respect to the reference MiRNA sequence. [<a href="https://en.wikipedia.org/wiki/IsomiR" class="uri">https://en.wikipedia.org/wiki/IsomiR</a>]</p>
<p>There are 5 different types of isomiRs:</p>
<ul>
<li>ref: perfect match to the sequence on the database</li>
<li>t3: nucleotides changes at 3’</li>
<li>t5: nucleotides changes at 5’</li>
<li>add: nucleotides addition at the 3’</li>
<li>mism: putatives SNPs</li>
</ul>
<div class="figure">
<img src="http://ars.els-cdn.com/content/image/1-s2.0-S0168952512001126-gr1.jpg">
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ids &lt;-<span class="st"> </span><span class="kw"><a href="../reference/bcbio.html">bcbio</a></span>(bcb, <span class="st">"isomirs"</span>)
<span class="kw"><a href="http://www.rdocumentation.org/packages/isomiRs/topics/isoPlot">isoPlot</a></span>(ids, <span class="dt">type =</span> <span class="st">"all"</span>, <span class="dt">column =</span> <span class="st">"country"</span>)</code></pre></div>
<div class="figure">
<img src="qc-example_files/figure-html/isomiRs-1.png" alt="Proportion of isomiRs counting reads (left) or unique sequences (right)." width="700"><p class="caption">
Proportion of isomiRs counting reads (left) or unique sequences (right).
</p>
</div>
</div>
<div id="small-rna-clusters" class="section level2">
<h2 class="hasAnchor">
<a href="#small-rna-clusters" class="anchor"></a>Small RNA clusters</h2>
<p>We use <a href="http://seqcluster.readthedocs.io/getting_started.html">seqcluster</a> to identify any type of small RNA. It generates a list of clusters of small RNA sequences, their genome location, their annotation and the abundance in all the sample of the project. The next figure explain the algorithm used for that and the way to use it for differential expression analyses.</p>
<div style="width:150px; height=150px">
<div class="figure">
<img src="http://seqcluster.readthedocs.io/_images/seqcluster.png" style="width:50.0%">
</div>
</div>
<p>This file generated by seqcluster, <code>seqcluster.db</code>, can be used with the page <code>reader.html</code> after downloading from <a href="https://github.com/lpantano/seqclusterViz/archive/master.zip">here</a>.</p>
<p>See example:</p>
<p><a href="https://youtu.be/Zjzte8n2-Sg" title="Example"><img src="http://i.makeagif.com/media/7-03-2016/M0GjW2.gif" alt="IMAGE ALT TEXT"></a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plots-smallrna.html">bcbSmallCluster</a></span>(bcb)</code></pre></div>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<pre><code>## Warning: Removed 92 rows containing non-finite values (stat_boxplot).</code></pre>
<div class="figure">
<img src="qc-example_files/figure-html/plot_srna_clusters-1.png" alt="Reads kept after each filering step(top left). Proportion of sequences detected (top right). SmallRNA proportions (bottom)." width="576"><p class="caption">
Reads kept after each filering step(top left). Proportion of sequences detected (top right). SmallRNA proportions (bottom).
</p>
</div>
<div id="clustering-of-samples-1" class="section level3">
<h3 class="hasAnchor">
<a href="#clustering-of-samples-1" class="anchor"></a>Clustering of samples</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data =<span class="st"> </span><span class="kw"><a href="../reference/plotsSmallPCA.html">bcbSmallPCA</a></span>(bcb, <span class="st">"cluster"</span>)
color_by =<span class="st"> </span><span class="kw">metadata</span>(bcb)[[<span class="st">"interesting_groups"</span>]]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">palette &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/circlize/topics/colorRamp2">colorRamp2</a></span>(<span class="kw">seq</span>(<span class="kw">min</span>(data[[<span class="st">"counts"</span>]]),
                          <span class="kw">max</span>(data[[<span class="st">"counts"</span>]]), <span class="dt">length =</span> <span class="dv">3</span>),
                      <span class="kw">c</span>(<span class="st">"blue"</span>, <span class="st">"#EEEEEE"</span>, <span class="st">"orange"</span>), <span class="dt">space =</span> <span class="st">"RGB"</span>)
th &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/ComplexHeatmap/topics/HeatmapAnnotation">HeatmapAnnotation</a></span>(<span class="dt">df =</span> data[[<span class="st">"annotation"</span>]],
                        <span class="dt">col =</span> <span class="kw"><a href="http://www.rdocumentation.org/packages/DEGreport/topics/degColors">degColors</a></span>(data[[<span class="st">"annotation"</span>]]))
<span class="kw"><a href="http://www.rdocumentation.org/packages/ComplexHeatmap/topics/Heatmap">Heatmap</a></span>(data[[<span class="st">"counts"</span>]],
        <span class="dt">col =</span> palette,
        <span class="dt">top_annotation =</span> th,
        <span class="dt">clustering_method_rows =</span> <span class="st">"ward.D"</span>,
        <span class="dt">clustering_distance_columns =</span> <span class="st">"kendall"</span>,
        <span class="dt">clustering_method_columns =</span> <span class="st">"ward.D"</span>,
        <span class="dt">show_row_names =</span> <span class="ot">FALSE</span>,
        <span class="dt">show_column_names =</span> <span class="kw">ncol</span>(data[[<span class="st">"counts"</span>]]) <span class="op">&lt;</span><span class="st"> </span><span class="dv">50</span>)</code></pre></div>
<p><img src="qc-example_files/figure-html/cluster-heatmap-1.png" width="700"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="http://www.rdocumentation.org/packages/DEGreport/topics/degPCA">degPCA</a></span>(data[[<span class="st">"counts"</span>]], data[[<span class="st">"annotation"</span>]],
            <span class="dt">condition =</span> color_by)</code></pre></div>
<p><img src="qc-example_files/figure-html/cluster-pca-1.png" width="700"></p>
</div>
</div>
</div>
<div id="session" class="section level1">
<h1 class="hasAnchor">
<a href="#session" class="anchor"></a>Session</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/devtools/topics/session_info">session_info</a></span>()</code></pre></div>
<pre><code>##  setting  value                       
##  version  R version 3.4.4 (2018-03-15)
##  system   x86_64, darwin15.6.0        
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       America/New_York            
##  date     2018-06-06                  
## 
##  package              * version    date      
##  acepack                1.4.1      2016-10-29
##  annotate               1.56.0     2017-10-31
##  AnnotationDbi          1.40.0     2017-10-31
##  assertive.base         0.0-7      2016-12-30
##  assertive.sets         0.0-3      2016-12-30
##  assertthat             0.2.0      2017-04-11
##  backports              1.1.2      2017-12-13
##  base                 * 3.4.4      2018-03-15
##  base64enc              0.1-3      2015-07-28
##  bcbioSmallRna        * 0.0.1      2018-06-06
##  bindr                  0.1.1.9000 2018-06-06
##  bindrcpp             * 0.2.2.9000 2018-06-06
##  Biobase              * 2.38.0     2017-10-31
##  BiocGenerics         * 0.24.0     2017-10-31
##  BiocParallel           1.12.0     2017-10-31
##  BiocStyle            * 2.5.41     2017-10-12
##  bit                    1.1-12     2014-04-09
##  bit64                  0.9-7      2017-05-08
##  bitops                 1.0-6      2013-08-17
##  blob                   1.1.0      2017-06-17
##  caTools                1.17.1     2014-09-10
##  checkmate              1.8.5      2017-10-24
##  circlize             * 0.4.3      2017-12-20
##  cluster                2.0.6      2017-03-10
##  colorspace             1.3-2      2016-12-14
##  commonmark             1.4        2017-09-01
##  compiler               3.4.4      2018-03-15
##  ComplexHeatmap       * 1.17.1     2017-10-25
##  ConsensusClusterPlus   1.42.0     2017-10-31
##  cowplot                0.9.2      2017-12-17
##  crayon                 1.3.4      2017-09-16
##  data.table             1.11.4     2018-05-27
##  datasets             * 3.4.4      2018-03-15
##  DBI                    0.8        2018-03-02
##  DEGreport            * 1.17.2     2018-06-06
##  DelayedArray         * 0.4.1      2017-11-07
##  desc                   1.1.1      2017-08-03
##  DESeq2                 1.18.1     2017-11-12
##  devtools               1.13.4     2017-11-09
##  digest                 0.6.15     2018-01-28
##  DiscriMiner          * 0.1-29     2013-11-15
##  dplyr                  0.7.5      2018-05-19
##  edgeR                  3.20.9     2018-02-27
##  evaluate               0.10.1     2017-06-24
##  foreign                0.8-69     2017-06-22
##  Formula                1.2-2      2017-07-10
##  fs                     1.2.2      2018-03-21
##  gdata                  2.18.0     2017-06-06
##  genefilter             1.60.0     2017-10-31
##  geneplotter            1.56.0     2017-10-31
##  GenomeInfoDb         * 1.14.0     2017-10-31
##  GenomeInfoDbData       0.99.1     2017-10-13
##  GenomicRanges        * 1.30.3     2018-02-26
##  GetoptLong             0.1.6      2017-03-07
##  GGally                 1.4.0      2018-05-17
##  ggdendro               0.1-20     2016-04-27
##  ggplot2              * 2.2.1      2016-12-30
##  ggrepel                0.8.0      2018-05-09
##  GlobalOptions          0.0.12     2017-05-21
##  glue                   1.2.0      2017-10-29
##  gplots                 3.0.1      2016-03-30
##  graphics             * 3.4.4      2018-03-15
##  grDevices            * 3.4.4      2018-03-15
##  grid                 * 3.4.4      2018-03-15
##  gridExtra              2.3        2017-09-09
##  gtable                 0.2.0      2016-02-26
##  gtools                 3.5.0      2015-05-29
##  highr                  0.6        2016-05-09
##  Hmisc                  4.1-1      2018-01-03
##  hms                    0.3        2016-11-22
##  htmlTable              1.11.2     2018-01-20
##  htmltools              0.3.6      2017-04-28
##  htmlwidgets            1.0        2018-01-20
##  httpuv                 1.3.5      2017-07-04
##  IRanges              * 2.12.0     2017-10-31
##  isomiRs              * 1.9.1      2018-06-06
##  janitor                1.0.0      2018-03-22
##  KernSmooth             2.23-15    2015-06-29
##  knitr                * 1.20       2018-02-20
##  labeling               0.3        2014-08-23
##  lattice                0.20-35    2017-03-25
##  latticeExtra           0.6-28     2016-02-09
##  lazyeval               0.2.1      2017-10-29
##  limma                  3.34.9     2018-02-22
##  locfit                 1.5-9.1    2013-04-20
##  logging                0.7-103    2013-04-12
##  magrittr               1.5        2014-11-22
##  MASS                   7.3-49     2018-02-23
##  Matrix                 1.2-14     2018-04-09
##  matrixStats          * 0.52.2     2017-04-14
##  memoise                1.1.0      2017-04-21
##  methods              * 3.4.4      2018-03-15
##  mime                   0.5        2016-07-07
##  mnormt                 1.5-5      2016-10-15
##  MultiAssayExperiment   1.4.9      2018-01-08
##  munsell                0.4.3      2016-02-13
##  nlme                   3.1-131.1  2018-02-16
##  nnet                   7.3-12     2016-02-02
##  Nozzle.R1              1.1-1      2013-05-15
##  parallel             * 3.4.4      2018-03-15
##  pillar                 1.1.0      2018-01-14
##  pkgconfig              2.0.1      2017-03-21
##  pkgdown                1.0.0      2018-05-03
##  plyr                   1.8.4      2016-06-08
##  psych                  1.8.4      2018-05-06
##  purrr                  0.2.5      2018-05-29
##  R6                     2.2.2      2017-06-17
##  RColorBrewer           1.1-2      2014-12-07
##  Rcpp                   0.12.17    2018-05-18
##  RCurl                  1.95-4.10  2018-01-04
##  readr                  1.1.1      2017-05-16
##  reshape                0.8.7      2017-08-06
##  reshape2               1.4.3      2017-12-11
##  rjson                  0.2.19     2018-05-18
##  rlang                  0.2.0.9001 2018-06-06
##  rmarkdown              1.6.0.9008 2017-10-31
##  roxygen2               6.0.1.9000 2017-10-31
##  rpart                  4.1-13     2018-02-23
##  rprojroot              1.2        2017-01-16
##  RSQLite                2.0        2017-06-19
##  rstudioapi             0.7        2017-09-07
##  S4Vectors            * 0.16.0     2017-10-31
##  scales                 0.5.0      2017-08-24
##  shape                  1.4.4      2018-02-07
##  shiny                  1.0.5      2017-08-23
##  shinydashboard         0.6.1      2017-06-14
##  splines                3.4.4      2018-03-15
##  stats                * 3.4.4      2018-03-15
##  stats4               * 3.4.4      2018-03-15
##  stringi                1.2.2      2018-05-02
##  stringr                1.3.1      2018-05-10
##  SummarizedExperiment * 1.8.1      2017-12-19
##  survival               2.41-3     2017-04-04
##  tibble                 1.4.2      2018-01-22
##  tidyr                  0.8.1      2018-05-18
##  tidyselect             0.2.4      2018-02-26
##  tools                  3.4.4      2018-03-15
##  utils                * 3.4.4      2018-03-15
##  withr                  2.1.2      2018-03-15
##  XML                    3.98-1.10  2018-02-19
##  xml2                   1.1.1      2017-01-24
##  xtable                 1.8-2      2016-02-05
##  XVector                0.18.0     2017-10-31
##  yaml                   2.1.19     2018-05-01
##  zlibbioc               1.24.0     2017-10-31
##  source                             
##  cran (@1.4.1)                      
##  cran (@1.56.0)                     
##  cran (@1.40.0)                     
##  cran (@0.0-7)                      
##  cran (@0.0-3)                      
##  CRAN (R 3.4.0)                     
##  cran (@1.1.2)                      
##  local                              
##  cran (@0.1-3)                      
##  local (lpantano/bcbioSmallRna@NA)  
##  Github (krlmlr/bindr@b6e6fd6)      
##  Github (krlmlr/bindrcpp@bd5ae73)   
##  cran (@2.38.0)                     
##  cran (@0.24.0)                     
##  cran (@1.12.0)                     
##  Bioconductor                       
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.0)                     
##  cran (@1.0-6)                      
##  CRAN (R 3.4.0)                     
##  cran (@1.17.1)                     
##  cran (@1.8.5)                      
##  cran (@0.4.3)                      
##  CRAN (R 3.4.4)                     
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.1)                     
##  local                              
##  cran (@1.17.1)                     
##  cran (@1.42.0)                     
##  cran (@0.9.2)                      
##  CRAN (R 3.4.1)                     
##  cran (@1.11.4)                     
##  local                              
##  cran (@0.8)                        
##  local (lpantano/DEGreport@c89180e) 
##  cran (@0.4.1)                      
##  CRAN (R 3.4.1)                     
##  Bioconductor                       
##  cran (@1.13.4)                     
##  cran (@0.6.15)                     
##  cran (@0.1-29)                     
##  cran (@0.7.5)                      
##  cran (@3.20.9)                     
##  cran (@0.10.1)                     
##  CRAN (R 3.4.4)                     
##  cran (@1.2-2)                      
##  CRAN (R 3.4.4)                     
##  cran (@2.18.0)                     
##  cran (@1.60.0)                     
##  cran (@1.56.0)                     
##  cran (@1.14.0)                     
##  Bioconductor                       
##  cran (@1.30.3)                     
##  cran (@0.1.6)                      
##  cran (@1.4.0)                      
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.0)                     
##  cran (@0.8.0)                      
##  cran (@0.0.12)                     
##  cran (@1.2.0)                      
##  cran (@3.0.1)                      
##  local                              
##  local                              
##  local                              
##  CRAN (R 3.4.1)                     
##  CRAN (R 3.4.0)                     
##  cran (@3.5.0)                      
##  cran (@0.6)                        
##  cran (@4.1-1)                      
##  CRAN (R 3.4.0)                     
##  cran (@1.11.2)                     
##  cran (@0.3.6)                      
##  cran (@1.0)                        
##  CRAN (R 3.4.1)                     
##  cran (@2.12.0)                     
##  local (lpantano/isomiRs@NA)        
##  cran (@1.0.0)                      
##  CRAN (R 3.4.4)                     
##  cran (@1.20)                       
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.4)                     
##  cran (@0.6-28)                     
##  cran (@0.2.1)                      
##  cran (@3.34.9)                     
##  cran (@1.5-9.1)                    
##  cran (@0.7-103)                    
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.4)                     
##  CRAN (R 3.4.4)                     
##  cran (@0.52.2)                     
##  CRAN (R 3.4.0)                     
##  local                              
##  CRAN (R 3.4.0)                     
##  cran (@1.5-5)                      
##  cran (@1.4.9)                      
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.4)                     
##  CRAN (R 3.4.4)                     
##  cran (@1.1-1)                      
##  local                              
##  cran (@1.1.0)                      
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.3)                     
##  CRAN (R 3.4.0)                     
##  cran (@1.8.4)                      
##  cran (@0.2.5)                      
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.0)                     
##  cran (@0.12.17)                    
##  cran (@1.95-4.)                    
##  CRAN (R 3.4.0)                     
##  cran (@0.8.7)                      
##  cran (@1.4.3)                      
##  cran (@0.2.19)                     
##  Github (tidyverse/rlang@ba4fb06)   
##  Github (rstudio/rmarkdown@ee11543) 
##  Github (klutometis/roxygen@bbf259d)
##  CRAN (R 3.4.4)                     
##  CRAN (R 3.4.0)                     
##  CRAN (R 3.4.1)                     
##  CRAN (R 3.4.1)                     
##  cran (@0.16.0)                     
##  CRAN (R 3.4.1)                     
##  cran (@1.4.4)                      
##  CRAN (R 3.4.1)                     
##  CRAN (R 3.4.0)                     
##  local                              
##  local                              
##  local                              
##  cran (@1.2.2)                      
##  cran (@1.3.1)                      
##  cran (@1.8.1)                      
##  CRAN (R 3.4.4)                     
##  cran (@1.4.2)                      
##  cran (@0.8.1)                      
##  cran (@0.2.4)                      
##  local                              
##  local                              
##  cran (@2.1.2)                      
##  cran (@3.98-1.)                    
##  CRAN (R 3.4.0)                     
##  cran (@1.8-2)                      
##  cran (@0.18.0)                     
##  cran (@2.1.19)                     
##  cran (@1.24.0)</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#exploratory-analysis">Exploratory analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#size-distribution">Size distribution</a></li>
      <li><a href="#mirna">miRNA</a></li>
      <li><a href="#isomirs">isomiRs</a></li>
      <li><a href="#small-rna-clusters">Small RNA clusters</a></li>
      </ul>
</li>
      <li><a href="#session">Session</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Lorena Pantano, Michael J. Steinbaugh, Harvard Chan Bioinformatics Core.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
